src/gpu/intel/gemm/host_scalars.hpp

```diff
@@ -25,22 +25,89 @@ namespace gpu {
 namespace intel {
 namespace gemm {
 
-// Get value of host side scalar from storage and convert to float
+// Get value of host side scalar from storage
 
-template <typename ScalarType>
-status_t get_scalar_value_as_float(float &scalar_value,
+template <typename ScalarType, typename ResultType>
+status_t get_scalar_value_as(ResultType &scalar_value,
         const host_scalar_memory_storage_t *scale_storage) {
     ScalarType value = 0;
     status_t status = scale_storage->get_scalar_value(&value, sizeof(value));
     assert(status == status::success);
     if (status != status::success) return status;
 
-    scalar_value = static_cast<float>(value);
+    scalar_value = static_cast<ResultType>(value);
     return status::success;
 }
 
-status_t maybe_get_scale_as_float(
-        const memory_storage_t &scale_storage, float &scalar_value);
+inline status_t maybe_get_host_scalar_value(
+        const memory_storage_t &scale_storage, float &scalar_value) {
+    using namespace data_type;
+    const host_scalar_memory_storage_t *scalar_storage
+            = utils::downcast<const host_scalar_memory_storage_t *>(
+                    &scale_storage);
+    status_t status = status::success;
+
+    switch ((int)scalar_storage->data_type()) {
+        case f32:
+            status = get_scalar_value_as<float, float>(
```

Comment:
Since `scalar_value` is a float, I think the compiler should be able to deduce the 2nd template parameter and you can drop it.